src/utils.lua

```diff
@@ -783,7 +783,9 @@ function utils.undernameForName(name)
 	end
 
 	local baseName = utils.baseNameForName(name)
-	return string.gsub(name:reverse(), baseName:reverse() .. "_", "", 1):reverse()
+	-- Escape the pattern to treat it as literal text, not a pattern
+	local pattern = string.gsub(baseName:reverse() .. "_", "([%-%.%+%[%]%(%)%$%^%%%?%*])", "%%%1")
```

Comment:
Readability here is pretty bad and this seems unnecessarily complicated. I've got an idea I'll push for ya now that you've got the tests in place...

Reply:
492fe1b

Reply:
And also d120437